e2e/tests/121-Portal/ViewPaProfilePage/ViewActivityOverviewMessageTab.spec.ts

```diff
@@ -0,0 +1,108 @@
+import HomePage from '@121-e2e/pages/Home/HomePage';
+import LoginPage from '@121-e2e/pages/Login/LoginPage';
+import RegistrationDetails from '@121-e2e/pages/RegistrationDetails/RegistrationDetailsPage';
+import TableModule from '@121-e2e/pages/Table/TableModule';
+import {
+  amountVisa,
+  paymentNrVisa,
+  programIdVisa,
+  registrationVisa as registrationVisaDefault,
+} from '@121-service/seed-data/mock/visa-card.data';
+import NLRCProgram from '@121-service/seed-data/program/program-nlrc-ocw.json';
+import { RegistrationStatusEnum } from '@121-service/src/registration/enum/registration-status.enum';
+import { SeedScript } from '@121-service/src/scripts/seed-script.enum';
+import { ProgramPhase } from '@121-service/src/shared/enum/program-phase.enum';
+import { StatusEnum } from '@121-service/src/shared/enum/status.enum';
+import {
+  changePhase,
+  doPayment,
+  waitForPaymentTransactionsToComplete,
+} from '@121-service/test/helpers/program.helper';
+import {
+  awaitChangePaStatus,
+  importRegistrations,
+} from '@121-service/test/helpers/registration.helper';
+import {
+  getAccessToken,
+  resetDB,
+} from '@121-service/test/helpers/utility.helper';
+import { test } from '@playwright/test';
+import data from '../../../../interfaces/Portal/src/assets/i18n/en.json';
+
+let accessToken: string;
+
+test.beforeEach(async ({ page }) => {
+  // Set WhatsApp-number for ALL tests in this suite only
+  const registrationVisa = {
+    ...registrationVisaDefault,
+    whatsappPhoneNumber: registrationVisaDefault.phoneNumber,
+  };
+
+  await resetDB(SeedScript.nlrcMultiple);
+  accessToken = await getAccessToken();
+
+  await changePhase(
+    programIdVisa,
+    ProgramPhase.registrationValidation,
+    accessToken,
+  );
+  await changePhase(programIdVisa, ProgramPhase.inclusion, accessToken);
+  await changePhase(programIdVisa, ProgramPhase.payment, accessToken);
+
+  // Arrange
+  await importRegistrations(programIdVisa, [registrationVisa], accessToken);
+  await awaitChangePaStatus(
+    programIdVisa,
+    [registrationVisa.referenceId],
+    RegistrationStatusEnum.included,
+    accessToken,
+  );
+  const paymentReferenceIds = [registrationVisa.referenceId];
+
+  // Act
+  await doPayment(
+    programIdVisa,
+    paymentNrVisa,
+    amountVisa,
+    paymentReferenceIds,
+    accessToken,
+  );
+
+  await waitForPaymentTransactionsToComplete(
+    programIdVisa,
+    paymentReferenceIds,
+    accessToken,
+    3001,
+    Object.values(StatusEnum),
+  );
+
```

Comment:
- I think I heard you mention during our call that this setup part needs to happen in a lot of test cases
- We also ran into this same code duplication problem in our API tests ;)
- Therefore we created some helper methods: you could replace this setup part with the seedPaidRegistrations or the seedIncludedRegistrations method from services/121-service/test/helpers/registration.helper.ts
- I think this can also be applied to some other playwright test cases

Reply:
I'll take a fix it is indeed very valuable to not repeat same thing over and over again.

Reply:
I have fixed this this and I'd like to address more improvements like this in a separate PR not to bloat this one with too many changes. Thanks a lot.